libcst/metadata/scope_provider.py

```diff
@@ -721,6 +722,12 @@ def visit_Annotation(self, node: cst.Annotation) -> Optional[bool]:
     def leave_Annotation(self, original_node: cst.Annotation) -> None:
         self.__in_annotation.discard(original_node)
 
+    def visit_Assign_value(self, node: cst.Assign) -> None:
+        self.__ignore_annotation += 1
+
+    def leave_Assign_value(self, node: cst.Assign) -> None:
+        self.__ignore_annotation -= 1
+
```

Comment:
Handle Assign differently doesn't seem right.
We should only consider the is_annotation for access in Annotation node (`__in_annotation` already takes case that).